### PR TITLE
Stabilizing tests and build pipeline

### PR DESCRIPTION
**Pull request by nw**
## Summary

Prepping repo to be made private

* Fixed tests
* Added code coverage
* Updated Jenkins
* removed travis webhooks and config
* removed older code, tools and examples
* bumped auth0-logging-common

## Checklist

- [x] Has automated tests
- [x] Compatible with Appliance
- [x] Has no dependencies to other PRs
- [x] Changelog updated (Use https://github.com/auth0/lock/blob/master/CHANGELOG.md as example)
- [x] Readme file updated to document changes

## Reminder checklist
- [ ] Version bumped

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,116 +1,133 @@
...
+    }
+  }
+}
+
+def runTests(String version) {
   try {
-    sh "npm test"
+    def npmCmd = getNpm(version)
+
+    if (version == '10') {
```
Comment (dafortune) [inline Jenkinsfile]: why only on version 10?

minor: maybe move them into a var or add a comment so the intent is a clearer?

File: Jenkinsfile [hunk 2]
```diff
@@ -1,116 +1,133 @@
...
+
     githubNotify context: 'jenkinsfile/auth0/tests', description: 'Tests passed', status: 'SUCCESS'
   } catch (error) {
     githubNotify context: 'jenkinsfile/auth0/tests', description: 'Tests failed', status: 'FAILURE'
     throw error
   }
 }
 
-def nodeVersions = ['8', '10', '12']
+def cleanInstall(version) {
```
Comment (dafortune) [inline Jenkinsfile]: these commands run on docker right?

File: Jenkinsfile (hunk 1 above)
Comment (nw) [reply Jenkinsfile]: @dafortune 

* supporting alias w/ npm for multiple hapi requires `npm@6`. v10, v12 are fine. It got a bit messy for testing v8. I see this as temp and didn't want to spend too much time.
* v12 - has deprecation notices that screw up the parsing of logs. Something to address just not a huge priority.
* Ideally I want to have multi-reporters, not a good story in mocha.

File: Jenkinsfile (hunk 2 above)
Comment (nw) [reply Jenkinsfile]: yes, they are all built in a docker image. `node:${version}`

**Pull request merged**